core/src/main/scala/kafka/server/ForwardingManager.scala

```diff
@@ -98,30 +100,40 @@ class ForwardingManagerImpl(
         val envelopeError = envelopeResponse.error()
         val requestBody = request.body[AbstractRequest]
 
-        val response = if (envelopeError != Errors.NONE) {
-          // An envelope error indicates broker misconfiguration (e.g. the principal serde
-          // might not be defined on the receiving broker). In this case, we do not return
-          // the error directly to the client since it would not be expected. Instead we
-          // return `UNKNOWN_SERVER_ERROR` so that the user knows that there is a problem
-          // on the broker.
-          debug(s"Forwarded request $request failed with an error in the envelope response $envelopeError")
-          requestBody.getErrorResponse(Errors.UNKNOWN_SERVER_ERROR.exception)
+        // Unsupported version indicates an incompatibility between controller and client API versions. The
```

Comment:
It may be helpful to mention that this can happen because the controller changed after a connection was established.